### PR TITLE
command for deploying application on gke from local

### PR DESCRIPTION
**Pull request by priyanshujain**

### CODE REVIEW

**Comment by priyanshujain** [comment]
Replace panic with error from the function.

File: cmd/main.go [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+	cmd.Stderr = os.Stderr
+	err = cmd.Run()
+	if err != nil {
+		panic(fmt.Sprintf("Error checking docker status: %s", err))
+	}
+
+	repoName := "yesbank"
+	dockerFilePath := "/Users/priyanshujain/go/src/github.com/okcredit/yesbank/Dockerfile"
+	repoPath := "/Users/priyanshujain/go/src/github.com/okcredit/yesbank"
+	opsPath := "/Users/priyanshujain/go/src/github.com/okcredit/yesbank-ops"
```
Comment (jaihindh-okc) [inline cmd/main.go]: likely unintended.

File: proxy/proxy.go [hunk 2]
```diff
@@ -0,0 +1,343 @@
...
+		}
+		localPorts[connection.LocalPort] = true
+	}
+
+	// return list of local ports from localPorts map
+	var localPortsList []int
+	for localPort := range localPorts {
+		localPortsList = append(localPortsList, localPort)
+	}
+	return localPortsList, nil
```
Comment (jaihindh-okc) [inline proxy/proxy.go:45]: `localPorts` is a set of integers. The authors of Go recommend using maps with `struct{}` as the value-type as sets. Based on that, came up with [a generic set-type](https://github.com/okcredit/staff-link/blob/de9a46903337d2c6db1cd073a55da94abf37fa6a/service/utils/methods.go#L104-L128). Doing something like that makes the conversion to `localPortsList` more convenient as well.

**Review by jaihindh-okc — APPROVED** [review]
Looks great. Installing prerequisites like `gcloud` and `kubectl` is a bit too much for my taste. I'd have just checked for their existence and exited if they didn't exist.

**Comment by priyanshujain** [comment]
> Looks great. Installing prerequisites like `gcloud` and `kubectl` is a bit too much for my taste. I'd have just checked for their existence and exited if they didn't exist.

I agree that It might be too much in the context of `devcli`. It comes from my want for a less magical package manager than homebrew and my time spent debugging `gcloud` config issues for others in past. I have a lot to share about package managers(especially brew). I will try to write about it.

File: proxy/proxy.go (hunk 2 above)
Comment (priyanshujain) [reply proxy/proxy.go:45]: This is an interesting idea, an empty struct occupies zero bytes in memory. Here we don't care about the value in the set so it should work. Let me change the logic.

**Pull request merged**